pageserver/src/tenant/timeline.rs

```diff
@@ -2759,6 +2782,8 @@ impl Timeline {
                 image_layers.push(image_layer);
             }
         }
+        // All wanted layers are taken in account by time_for_new_image_layer
+        *self.wanted_image_layers.lock().unwrap() = None;
```

Comment:
Because we do not hold the lock over multiple accesses to this lock, we could end up clearing here something which we haven't yet processed seen. Similarly between subsequent time_for_new_image_layer the contents could get changed at least looking at only this lock and not all of the other locks...

I think in that case it would be all right, just wasted allocations. However perhaps we are better to not reuse the allocation and allow allocator to reuse the memory instead, so no need to change that.

```suggestion
        // All wanted layers are taken in account by time_for_new_image_layer.
        // The watend_image_layers could get updated out of turn and we could
        // clear something which hasn't been looked at all. This is fine, because
        // next gc round any wanted would get added back in.
        *self.wanted_image_layers.lock().unwrap() = None;
```